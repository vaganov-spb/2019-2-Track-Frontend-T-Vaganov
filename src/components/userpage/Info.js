/* eslint-disable react/no-unused-state */
import React from 'react';
import { connect } from 'react-redux';
import  NameAndBioBox from './NameAndBioBox';
import infoStyles from '../../styles/ProfileInfo.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

function Info(props) {
	if (!props.isLoaded) {
		return <div />;
	}

	return (
		<div className={infoStyles.container}>
			<div className={infoStyles.img_place}>
				<div className={infoStyles.img_box}>
					<img className={infoStyles.img_content} alt="" src={props.user.src}/>
					<img className={infoStyles.watermark} alt="" src="https://www.stickpng.com/assets/thumbs/584abf1a2912007028bd9333.png"/>
				</div>
			</div>
			<NameAndBioBox/>
		</div>
	);
}

const mapStateToProps = (state) => ({
	user: state.userinfo.info,
	isLoaded: state.userinfo.isLoaded,
	error: state.userinfo.error
});

export default connect(mapStateToProps)(Info);