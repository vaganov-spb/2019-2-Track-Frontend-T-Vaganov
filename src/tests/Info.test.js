import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Info from '../components/userpage/Info';

const mockStore = configureStore();

describe('Info renders and act correctly', () => {
	let store;
	let component;
	let elem;

	beforeEach(() => {
		store = mockStore({
			userinfo: {
				info: {
					Name: 'Borat',
					Nick: 'Sagdiev',
					Bio: 'Pasha Technik',
					src: 'img'
				},
				isLoaded: true,
				error: null, 
			}
		});
        
		component = renderer.create(
			<Provider store={store}>
				<MemoryRouter>
					<Info/>
				</MemoryRouter>  
			</Provider>
		);
         
		elem = component.root;
	});

	it ('should render normally', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('should have right src', () => {
		const img = elem.findByProps({className: 'img_content'});
		expect(img.props.src).toBe('img');
	});
});