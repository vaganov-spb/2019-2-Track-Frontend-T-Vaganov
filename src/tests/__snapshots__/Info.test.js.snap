// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders and act correctly should render normally 1`] = `
<div
  className="container"
>
  <div
    className="img_place"
  >
    <div
      className="img_box"
    >
      <img
        alt=""
        className="img_content"
        src="img"
      />
      <img
        alt=""
        className="watermark"
        src="https://www.stickpng.com/assets/thumbs/584abf1a2912007028bd9333.png"
      />
    </div>
  </div>
  <div
    className="user_fields"
  >
    <div
      className="wrapper"
    >
      <div
        className="field_name"
      >
         
        Name
         
      </div>
      <textarea
        className="name"
        maxLength="25"
        minLength="5"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter your name!"
        rows="1"
        value="Borat"
      />
      <div
        className="userlen"
      >
         
         
      </div>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="field_name"
      >
         
        Nick
         
      </div>
      <textarea
        className="name"
        maxLength="25"
        minLength="5"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Enter your username!"
        rows="1"
        value="Sagdiev"
      />
      <div
        className="userlen"
      >
         
        Minimum lenght is 5 characters
         
      </div>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="field_name"
      >
         
        Bio
         
      </div>
      <textarea
        className="name"
        maxLength="75"
        minLength="5"
        onChange={[Function]}
        onKeyPress={[Function]}
        rows="3"
        value="Pasha Technik"
      />
      <div
        className="userlen"
      >
         
        Any details about you
         
      </div>
    </div>
  </div>
</div>
`;
